src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -168,119 +177,186 @@ public InventorySyncStatistics getStatistics() {
+     *
+     * @param oldInventories list of old {@link InventoryEntry} instances
+     * @param draft new {@link InventoryEntryDraft} instance where supply channel reference, if present, points to
+     *              channel in target CTP project
+     * @return {@link Optional} that may contain {@link InventoryEntry} found in {@code oldInventories} that correspond
+     *      to {@code draft} or empty {@link Optional} if no such entry was found
+     */
+    private Optional<InventoryEntry> getCorrespondingEntry(@Nonnull final List<InventoryEntry> oldInventories,
+                                                           @Nonnull final InventoryEntryDraft draft) {
+        return oldInventories.stream()
+            .filter(oldInventory -> oldInventory.getSku().equals(draft.getSku()))
+            .filter(oldInventory -> draft.getSupplyChannel() == null
+                ? (oldInventory.getSupplyChannel() == null)
+                : ((oldInventory.getSupplyChannel() != null)
+                && oldInventory.getSupplyChannel().getId().equals(draft.getSupplyChannel().getId())))
+            .findFirst();
+    }
+
+    /**
+     * Returns {@link CompletionStage} instance which may contain mapping of {@link ChannelKeyExtractor} to
      * {@link InventoryEntry} of instances existing in a CTP project. Instances are fetched from API by skus, that
      * corresponds to skus present in {@code drafts}. If fetching existing instances results in exception then
      * returned {@link CompletionStage} contains such exception.
      *
      * @param drafts {@link List} of drafts
-     * @return {@link CompletionStage} instance which contains either {@link Map} of {@link SkuChannelKeyTuple} to
+     * @return {@link CompletionStage} instance which contains either {@link Map} of {@link ChannelKeyExtractor} to
      *      {@link InventoryEntry} of instances existing in a CTP project that correspond to passed {@code drafts} by
      *      sku comparision, or exception occurred during fetching existing inventory entries
      */
-    private CompletionStage<Map<SkuChannelKeyTuple, InventoryEntry>> fetchExistingInventories(final
-                                                                                              List<InventoryEntryDraft>
-                                                                                                  drafts) {
+    private CompletionStage<List<InventoryEntry>> fetchExistingInventories(@Nonnull final List<InventoryEntryDraft>
+                                                                               drafts) {
         final Set<String> skus = extractSkus(drafts);
-        return inventoryService.fetchInventoryEntriesBySkus(skus)
-            .thenApply(existingEntries -> existingEntries.stream().collect(toMap(SkuChannelKeyTuple::of,
-                entry -> entry)));
+        return inventoryService.fetchInventoryEntriesBySkus(skus);
```

Comment:
move exception handling here if it is the same concern, don't wrap it in caller method